typhos/alarm.py

```diff
@@ -0,0 +1,380 @@
+"""
+Module to define alarm summary frameworks and widgets.
+"""
+from functools import partial
+import logging
+import os
+
+from ophyd.device import Kind
+from ophyd.signal import EpicsSignalBase
+from pydm.widgets.base import PyDMPrimitiveWidget
+from pydm.widgets.channel import PyDMChannel
+from pydm.widgets.drawing import (PyDMDrawing, PyDMDrawingCircle,
+                                  PyDMDrawingRectangle, PyDMDrawingTriangle,
+                                  PyDMDrawingEllipse, PyDMDrawingPolygon)
+from qtpy import QtCore, QtWidgets
+
+from .plugins import register_signal
+from .utils import (channel_from_signal, get_all_signals_from_device,
+                    TyphosObject)
+from .widgets import HappiChannel
+
+
+logger = logging.getLogger(__name__)
+
+
+class KindLevel:
```

Comment:
Entirely optional, but consider using: https://github.com/zllentz/typhos/blob/d9370c48e0c1c2c6356191fe9f1569c213e2a0d2/typhos/display.py#L24-L32

We had talked about this the other day, and I _knew_ there was something in some package, but I couldn't find it then. Benefit is that you can use `IntEnum` for Python friendliness but also support Q_ENUM without issue.

Reply:
This is worth doing, I'll edit appropriately